src/alchemlyb/estimators/ti_gaussian_quadrature.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+
+from .base import _EstimatorMixOut
+
+
+class TI_gq(BaseEstimator, _EstimatorMixOut):
+    """Thermodynamic integration (TI) with gaussian quadrature estimation.
+
+    Parameters
+    ----------
+
+    verbose : bool, optional
+        Set to True if verbose debug output is desired.
+
+    Attributes
+    ----------
+
+    delta_f_ : DataFrame
+        The estimated dimensionless free energy difference between each state.
+
+    d_delta_f_ : DataFrame
+        The estimated statistical uncertainty (one standard deviation) in
+        dimensionless free energy differences.
+
+    states_ : list
+        Lambda states for which free energy estimation were obtained.
+
+    dhdl : DataFrame
+        The estimated dhdl of each state.
+
+
+    .. versionchanged:: 2.0.0
+       `delta_f_`, `d_delta_f_`, `states_` are view of the original object.
+
+    """
+
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+    def fit(self, dHdl):
+        """
+        Compute free energy differences between each state by integrating
+        dHdl across lambda values.
+
+        Parameters
+        ----------
+        dHdl : DataFrame
+            dHdl[n,k] is the potential energy gradient with respect to lambda
+            for each configuration n and lambda k.
+
+        """
+
+        # sort by state so that rows from same state are in contiguous blocks,
+        # and adjacent states are next to each other
+        dHdl = dHdl.sort_index(level=dHdl.index.names[1:])
+
+        # obtain the mean and variance of the mean for each state
+        # variance calculation assumes no correlation between points
+        # used to calculate mean
+        means = dHdl.groupby(level=dHdl.index.names[1:]).mean()
+        variances = np.square(dHdl.groupby(level=dHdl.index.names[1:]).sem())
+
+        # extract the lambda values used in the simulations
+        lambdas = means.reset_index()[means.index.names[:]].iloc[:].values.sum(axis=1)
```

Comment:
Is this safe for the dataframe that has more than one lambda values?

Reply:
I changed this part to support multi-lambda situation. However, scaling multiple lambdas at the same time is not supported